### PR TITLE
Now using TimeSpan in Promise/Future

### LINKED ISSUE #1307: `wait` in Promise should accept TimeSpan instead of double
to improve readability and get rid of ugly constants

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1307

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/base/TimeSpan.ooc [hunk 1]
```diff
@@ -16,6 +16,9 @@ TimeSpan: cover {
 		this _ticks = DateTime timeToTicks(hour, minute, second, millisecond)
 	}
 	negate: func -> This { This new(-1 * this ticks) }
+	elapsedNanoseconds: func -> Long {
+		1_000_000 * this elapsedMilliseconds()
```
Comment (sebastianbaginski) [inline source/base/TimeSpan.ooc]: Why not `this ticks / DateTime nanosecondsPerTick` ?
Current one will return zero for ticks corresponding to less than 1 ms. (as `elapsedMilliseconds()` will round down to zero).

**Comment by sebastianbaginski** [comment]
> 16:49:17 ThreadPool (SIGSEGV) segmentation fault

:)

As for the PR, no comments other than the above one.

File: source/base/TimeSpan.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/base/TimeSpan.ooc]: Oh, I didn't know we had that, thanks.

**Comment by marcusnaslund** [comment]
Updated as you said, but with a multiplication sign though. And a test for nanoseconds.

**Comment by sebastianbaginski** [comment]
Oh of course, it is division in DateTime :)

**Pull request merged**